File: Data Structures & Algorithms/concatenation-of-array/submission-3.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        new = [0] * (2 * len(nums))

        for i in range(len(nums)):
            new[i] = nums[i]
        
        old = 0
        for i in range(len(nums), len(new)):
            new[i] = nums[old]
            old += 1
        return new

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-3.ts
class Solution {
    /**
     * @param {number} target
     * @param {number[]} nums
     * @return {number}
     */
    minSubArrayLen(target: number, nums: number[]): number {
        let len = nums.length + 1;
        let l = 0;
        let total = 0;
        for (let r = 0; r < nums.length; r++) {
            total += nums[r];
            while (total >= target) {
                len = Math.min(len, r - l + 1);
                total -= nums[l];
                l++;
            }
        }

        if (len === nums.length + 1) {
            return 0;
        }
        return len;

    }
}
